### ISSUE TITLE
Accessibility issue with labels in Registration Form course

### ISSUE THREAD

**Issue by bbsmooth** [freeCodeCamp/freeCodeCamp#46593]
### Describe the Issue

There are two related issues:

- The label for the file input does not use a `for` attribute and this is causing the NVDA screen reader on Windows and ORCA screen reader on linux to not announce the label text when the user tabs to the input. I have a [codepen demo](https://codepen.io/bbsmooth/pen/WNMBjVK) to illustrate this.
- It is considered a [best practice to use a `for` attribute on a label,](https://www.w3.org/WAI/tutorials/forms/labels/) even when the input is embedded in the label itself, because some screen readers may not read the label correctly without the `for` attribute (as demonstrated in the issue above). This is also noted in [step 30 of the Accessibility Quiz course](https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-accessibility-by-building-a-quiz/step-30).

All of the inputs on the form in this course are embedded in a `label` and none of the labels have a `for` attribute.

### Affected Page

All steps of the Registration Form course that have inputs embedded in labels.

### Your code

Current HTML for file input:

```
<label>Upload a profile picture: <input type="file" name="file" /></label>
```

### Expected behavior

Should be something like:

```
<label for="file-upload">Upload a profile picture: <input id="file-upload" type="file" name="file" /></label>
```

### System

- Device: Desktop
- OS: Windows 11
- Browser: Firefox
- Version: 101
- Screen Reader: NVDA
- Version: 2022.1

### Additional context

I'm only bringing this up because I thought it might be a good opportunity to teach campers that valid HTML isn't always as accessible as it could be and best practices should be followed because they help eliminate these types of edge cases.

**Comment by Sboonny** [comment]
The concept is introduced early on in https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-html-by-building-a-cat-photo-app/step-57 , I don't see any issues with this. I am okay with it being implemented 👍

**Comment by bbsmooth** [comment]
@ShaunSHamilton I think we would need to add more to the descriptions because we would need to tell the user to add an `id` to the `input` and a corresponding `for` attribute to the `label`. Or I suppose we could just automatically add them behind the scenes but I'm not sure that wouldn't cause more confusion than it's worth. As you know by now, I'm a big accessibility advocate and I will almost always argue that it is worth the time, effort, and space to make people aware of accessibility issues and best practices. But I do understand that you want to keep these as simple as possible and there are a lot of label/inputs in this course that would need to be changed.

I've said my two cents. If this isn't something that people want to do then feel free to close this issue.

**Comment by JohnathanTWebster** [comment]
Does this still need to be implemented? I know that the new course is out now so wanted to verify before looking further into this issue.

**Comment by Sboonny** [comment]
@JohnathanTWebster this issue is in the new RWD course

**Comment by JohnathanTWebster** [comment]
@Sboonny Okay, I can take a look. Since the concept is mentioned early on in the course I can start adding "for" in the labels.

**Comment by JohnathanTWebster** [comment]
Maybe I need further clarification. After going through all the "Learn Accessibility by Building a Quiz" steps I am unable to locate where "for" is missing in the labels. Starting from step 21 and throughout the curriculum the labels are all populating with the correct "for" descriptions. I have also run the site in accessibility mode, NVDA as I am using windows 10, and the narrator is correctly identifying the labels.
Is there something I am missing with this issue?

**Comment by JohnathanTWebster** [comment]
Yeah, I am gonna need a little help with where to start on this. Where is the HTML being formatted from Mark Down? I have spent the majority of today trying to figure that out but have gotten lost.

**Comment by Sboonny** [comment]
@JohnathanTWebster there is a ci tool that take care of this, you can follow the instruction mentioned in this pull request.

Other than that, it's best to use search function on your Text editor.

Here is the markdown file https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/14-responsive-web-design-22/learn-accessibility-by-building-a-quiz/6145e6eeaa66c605eb087fe9.md

**Comment by JohnathanTWebster** [comment]
@Sboonny thank you. I will look into this further. I checked the documentation site for how to "edit" but there were only steps on how to create new challenges.

**Comment by JohnathanTWebster** [comment]
Before moving forward, as it was mentioned in the initial post, here are a couple samples I would like to review. The first sample is with 1:1 naming meaning the same name as the entered text.

```  
<h1>Registration Form</h1>
    <p>Please fill out this form with the required information</p>
    <form action='https://register-demo.freecodecamp.org'>
      <fieldset>
        <label for="Enter-Your-First-Name">Enter Your First Name: <input id="Enter-Your-First-Name" type="text" required /></label>
        <label for="Enter-Your-First-Last">Enter Your Last Name: <input id="Enter-Your-First-Last" type="text" required /></label>
        <label for="Enter-Your-Email">Enter Your Email: <input id="Enter-Your-Email" type="email" required /></label>
        <label for="Create-a-New-Password">Create a New Password: <input id="Create-a-New-Password" type="password" pattern="[a-z0-5]{8,}" required /></label>
      </fieldset>
      <fieldset>
        <label for="Personal-Account"><input id="Personal-Account" type="radio" name="account-type" /> Personal Account</label>
        <label for="Business-Account"><input id="Business-Account" type="radio" name="account-type" /> Business Account</label>
        <label for="I-Accept-The-Terms-And-Conditions">
          <input id="I-Accept-The-Terms-And-Conditions" type="checkbox" required /> I accept the <a href="https://www.freecodecamp.org/news/terms-of-service/">terms and conditions</a>
			  </label>
```
Personally I am not a fan of the wordy formatting only because the auto wrapping for the site gets really crowded. Though, the descriptions are more informative.

The second example is less wordy/informative but wraps much better compared to the first option.

```
    <h1>Registration Form</h1>
    <p>Please fill out this form with the required information</p>
    <form action='https://register-demo.freecodecamp.org'>
      <fieldset>
        <label for="First-Name">Enter Your First Name: <input id="First-Name" type="text" required /></label>
        <label for="Last-Name">Enter Your Last Name: <input id="Last-Name" type="text" required /></label>
        <label for="Email">Enter Your Email: <input id="Email" type="email" required /></label>
        <label for="New-Password">Create a New Password: <input id="New-Password" type="password" pattern="[a-z0-5]{8,}" required /></label>
      </fieldset>
      <fieldset>
        <label for="Personal-Account"><input id="Personal-Account" type="radio" name="account-type" /> Personal Account</label>
        <label for="Business-Account"><input id="Business-Account" type="radio" name="account-type" /> Business Account</label>
        <label for="Terms-Conditions">
          <input id="Terms-Conditions" type="checkbox" required /> I accept the <a href="https://www.freecodecamp.org/news/terms-of-service/">terms and conditions</a>
			  </label>
```

And the final option is even less wordy  but is still better for screen wrapping. 

```
    <h1>Registration Form</h1>
    <p>Please fill out this form with the required information</p>
    <form action='https://register-demo.freecodecamp.org'>
      <fieldset>
        <label for="firstname">Enter Your First Name: <input id="firstname" type="text" required /></label>
        <label for="lastname">Enter Your Last Name: <input id="lastname" type="text" required /></label>
        <label for="email">Enter Your Email: <input id="email" type="email" required /></label>
        <label for="newpassword">Create a New Password: <input id="newpassword" type="password" pattern="[a-z0-5]{8,}" required /></label>
      </fieldset>
      <fieldset>
        <label for="personalaccount"><input id="personalaccount" type="radio" name="account-type" /> Personal Account</label>
        <label for="businessaccount"><input id="businessaccount" type="radio" name="account-type" /> Business Account</label>
        <label for="termsconditions">
          <input id="termsconditions" type="checkbox" required /> I accept the <a href="https://www.freecodecamp.org/news/terms-of-service/">terms and conditions</a>
			  </label>
```
I have not adjusted the descriptions as requested and would like some input on which styling to continue with.

**Comment by Sboonny** [comment]
I prefer the second but without capitalizing first latters, because it's the same style as the one used in previous steps.

**Comment by JohnathanTWebster** [comment]
Okay, I'll continue with the second without capitalizing the first letters
